### PR TITLE
dpdk: support table.apply().hit and table.apply().miss expression in …

### PR DESCRIPTION
**Pull request by hanw**
…if statement

### CODE REVIEW

File: testdata/p4_16_samples_outputs/psa-table-hit.p4.spec [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	default_action ifHit args none 
+	size 0
+}
+
+
+apply {
+	rx m.psa_ingress_input_metadata_ingress_port
+	extract h
+	table tbl_0
+	jmph LABEL_0END
```
Comment (mbudiu-vmw) [inline testdata/p4_16_samples_outputs/psa-table-hit.p4.spec:71]: is this the correct output or is it inverted?

**Review by mbudiu-vmw — APPROVED** [review]

File: testdata/p4_16_samples_outputs/psa-table-hit.p4.spec (hunk 1 above)
Comment (hanw) [reply testdata/p4_16_samples_outputs/psa-table-hit.p4.spec:71]: This is correct to my understanding, see comment from Crisitian below.

>“Jump if table lookup hit” (jmph) example:
table vxlan_table
//some instructions here, but none of them is “table”, as jmph and jmpnh will refer to the table from the last “table” instruction
jmph GO_HERE_ON_VXLAN_TABLE_HIT
//some instructions
GO_HERE_ON_VXLAN_TABLE_HIT:
//some instructions to execute on vxlan_table hit

**Comment by hesingh** [comment]
In the P4 code, I don't see a test for table.apply().miss?

File: testdata/p4_16_samples_outputs/psa-table-hit.p4.spec (hunk 1 above)
Comment (mbudiu-vmw) [reply testdata/p4_16_samples_outputs/psa-table-hit.p4.spec:71]: This still seems to me to be inverted. On hit you skip the tbl_ifHit application.

Comment (hanw) [reply testdata/p4_16_samples_outputs/psa-table-hit.p4.spec:71]: Thanks, I fixed the issue.

**Comment by hanw** [comment]
> 
> 
> In the P4 code, I don't see a test for table.apply().miss?

Thanks, added test for apply().miss.

**Pull request merged**